### PR TITLE
IGNITE-13051 still validate MVCC if cache will not start

### PR DESCRIPTION
**Pull request by timoninmaxim**
Validate and enable MVCC if cache will not start

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java [hunk 1]
```diff
@@ -1350,6 +1350,9 @@ else if (!crd && grp != null && grp.localStartVersion().equals(fut.initialVersio
         Collection<DynamicCacheDescriptor> descs
     ) throws IgniteCheckedException {
         IgniteInternalFuture<?> res = cachesRegistry.addUnregistered(descs);
+        for (DynamicCacheDescriptor d: descs) {
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java:1353]: newline required before "for"
see https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-SemanticUnits for details

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java [hunk 2]
```diff
@@ -322,6 +325,33 @@ public void testTransactionalSnapshotLimitations() throws Exception {
...
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testMvccEnabledForClientMode() throws Exception {
+        String attrName = "has_cache";
+        Object attrVal = Boolean.TRUE;
+
+        final IgniteEx crd = (IgniteEx) startGrid(getTestIgniteInstanceName(0), getConfiguration());
+
+        // Do not start cache on non-affinity node.
+        CacheConfiguration ccfg = defaultCacheConfiguration()
```
Comment (anton-vinogradov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java:340]: Do we really need to specify near, backups, and atomicity mode here?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java [hunk 3]
```diff
@@ -322,6 +325,33 @@ public void testTransactionalSnapshotLimitations() throws Exception {
...
+                .setBackups(0)
+                .setAtomicityMode(TRANSACTIONAL_SNAPSHOT);
+
+        final Ignite node = startGrid(getTestIgniteInstanceName(1), getConfiguration()
+                .setCacheConfiguration(ccfg)
+                .setUserAttributes(F.asMap(attrName, attrVal)));
+
+        checkTopology(2);
+
+        assertTrue(crd.context().coordinators().mvccEnabled());
```
Comment (anton-vinogradov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java:352]: The better case is to check every node.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java (hunk 2 above)
Comment (timoninmaxim) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java:340]: Hi, this test checks that MVCC mode is enabled, so atomicity mode is required for testing. Near must be set to null with enabled MVCC. Backups doesn't matter actually, I'll fix it

Comment (timoninmaxim) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java:340]: Hi, this test checks MVCC mode so atomicity mode is required. Near set is required as it must be disabled with MVCC mode. Backups dont matter actually. Fixed it

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java (hunk 3 above)
Comment (timoninmaxim) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/mvcc/CacheMvccConfigurationValidationTest.java:351]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java (hunk 1 above)
Comment (timoninmaxim) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java:1354]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java [hunk 4]
```diff
@@ -1351,6 +1351,10 @@ else if (!crd && grp != null && grp.localStartVersion().equals(fut.initialVersio
     ) throws IgniteCheckedException {
         IgniteInternalFuture<?> res = cachesRegistry.addUnregistered(descs);
 
+        for (DynamicCacheDescriptor d: descs) {
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java:1354]: oneliner for should not use braces

Comment (timoninmaxim) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java]: fixed

**Pull request merged**